src/BaseControllerV2.ts

```diff
@@ -134,7 +146,7 @@ export class BaseController<S extends Record<string, unknown>> {
    * @param listener - Callback triggered when state changes
    */
   subscribe(listener: Listener<S>) {
```

Comment:
Maybe we should remove `subscribe` and `unsubscribe` here instead :thinking: 

If we're expecting other controllers to subscribe via the controller messaging system directly, this method becomes superfluous. Similarly, in anyplace "above" the controllers (e.g. `background.js`, `metamask-controller.js`), the controller messaging system can be used directly instead.

Reply:
actually yeah that makes a lot of sense. That would require controllers to use the messenger to subscribe to "local" events, though. But that might actually make the messenger easier to understand if everything interacts with it.

Reply:
Great! They have been removed.